planning/behavior_path_planner/src/turn_signal_decider.cpp

```diff
@@ -65,28 +65,48 @@ std::pair<TurnIndicatorsCommand, double> TurnSignalDecider::getIntersectionTurnS
 
   // Get nearest intersection and decide turn signal
   double accumulated_distance = 0;
+  double turn_distance = intersection_search_distance_;
+  double path_distance = 0;
+  double judge_distance = 0;
 
   auto prev_point = path.points.front();
   auto prev_lane_id = lanelet::InvalId;
   for (const auto & path_point : path.points) {
-    accumulated_distance +=
-      tier4_autoware_utils::calcDistance3d(prev_point.point, path_point.point);
+    path_distance = tier4_autoware_utils::calcDistance3d(prev_point.point, path_point.point);
+    accumulated_distance += path_distance;
     prev_point = path_point;
     const double distance_from_vehicle_front =
       accumulated_distance - vehicle_pose_frenet.length - base_link2front_;
-    if (distance_from_vehicle_front < 0.0) {
-      continue;
+    if (distance_from_vehicle_front > intersection_search_distance_) {
+      return std::make_pair(turn_signal, distance);
     }
     // TODO(Horibe): Route Handler should be a library.
     for (const auto & lane : route_handler.getLaneletsFromIds(path_point.lane_ids)) {
+      if (lane.id() != path_point.lane_ids.back()) {
+        continue;
+      }
+      turn_distance = lane.attributeOr("turn_signal_distance", intersection_search_distance_);
+      if (turn_distance < min_turn_signal_distance_) {
+        turn_distance = min_turn_signal_distance_;
+      }
+      if (turn_distance < turn_signal_distance_threshold_) {
+        turn_distance = turn_signal_distance_threshold_;
+      }
+      if (turn_distance < path_distance) {
+        judge_distance = turn_distance - path_distance;
+        turn_distance += turn_signal_distance_threshold_;
+      } else {
+        judge_distance = 0.0;
+      }
```

Comment:
`judge_distance` should be defined in this scope, not outside this `for` loop.
Better to write like this
```
double judge_distance = 0.0;
if (turn_distance < path_distance) {
  judge_distance = turn_distance - path_distance;
  turn_distance += turn_signal_distance_threshold_;
}
```

Reply:
Fixed the above.

Reply:
Fixed the above.